active/0019-plugins.md

```diff
@@ -0,0 +1,126 @@
+
+All these changes we have made in the past are to provide better user experience as well as development experiences. But the flexibility of plugin applications and how they can be hosted in separate git repos still have their advantages and we should continue supporting it.
+
+The challenges ahead are
+
+- We want to continue supporting old plugins, but how to minimise the effort to migrate to 5.0.
+Although the hook points are still there, the config format is slightly different, and the schema is also different. (cuttlefish → HOCON)
+- How to provide a better management interfaces for external plugins e.g. management API or even UI.
+- Callbacks under the same hook point are ordered by their priority, however most of the plugins (internal plugins included) today are using the default value 0, meaning it's ordered by luck.
+
+Worth mentioning, since 4.3.2, EMQ X for the first supported drop-in installation of external plugins. That is, plugins can be compiled and packaged separately (instead of being released as a part of EMQ X package), see [Develop and deploy EMQ X plugin for Enterprise 4.3](https://emqx.atlassian.net/wiki/spaces/EMQX/blog/2021/05/23/168591472)
+
+## References
+
+- RabbitMQ: [Plugin Development Basics — RabbitMQ](https://www.rabbitmq.com/plugin-development.html)
+- Grafana: [Pie Chart](https://grafana.com/grafana/plugins/grafana-piechart-panel/)
+- WordPress: [WordPress Plugins](https://wordpress.org/plugins/)
+- RabbitMQ Plugin release packages: [Releases · rabbitmq/rabbitmq-management-exchange](https://github.com/rabbitmq/rabbitmq-management-exchange/releases)
+
+### High level requirements
+
+1. Hook points should be compatible with 4.x
+2. More clear ordering of all hook callbacks under one certain hook point, for example, internal hooks use priority 1000, and force external plugins to provide a priority number. If they wish to have it ordered before internal plugins they can choose to use a number smaller than 1000.
+3. All built-in features such as authentication and authorisation are not presented as plugins except for
+    1. LDAP auth
+
+### Basic steps to install an external plugin
+
+- Download compiled zip package
+- Upload to a specific dir
+- Execute a command to validate & install & enable the plugin
+
+### Manage plugins from Dashboard UI
+
+- Manage installation from Dashboard GUI
+    - upload
+    - install
+    - uninstall
+- Manage a list of installed plugins, supported actions:
+    - List view
+    - Show running status: running | stopped
+    - support actions: start | stop
+
+## Plugin metadata
+
+The plugin package should include metadata (in JSON format) to help identify, validate, or describe the package.
+
+- Name (same as the Erlang application name, it has to be globally unique)
+- Version
+- License (checked against a white list: apache2, mit, ...)
+- Author
+    - Name
+    - Contact
+- URL to source code
+- What functionalities (one or more of below)
```

Comment:
Are those like an enum of possibilities (that could change over versions), or just human-readable informative strings?

Reply:
it's only for documentation purpose. no strict rule.